ci-scripts/rhdh-setup/create_resource.sh

```diff
@@ -1,109 +1,130 @@
+  curl -k "$(backstage_url)/api/catalog/locations" -X POST -H 'Accept-Encoding: gzip, deflate, br' -H 'Content-Type: application/json' --data-raw '{"type":"url","target":"'"${upload_url}"'"}'
 }
 
-create_api() {
-  export grp_indx=$1	
+function create_api() {
+  export grp_indx=$1
   export api_indx=$2
-  cat template/component/api.template | envsubst '${grp_indx} ${api_indx}'>> api.yaml
+  cat $WORKDIR/template/component/api.template | envsubst '${grp_indx} ${api_indx}' >>$TMP_DIR/api.yaml
 }
 
-create_cmp() {
-  export grp_indx=$1	
+function create_cmp() {
+  export grp_indx=$1
   export cmp_indx=$2
-  cat template/component/component.template | envsubst '${grp_indx} ${cmp_indx}'>> component.yaml
+  cat $WORKDIR/template/component/component.template | envsubst '${grp_indx} ${cmp_indx}' >>$TMP_DIR/component.yaml
 }
 
-create_group() {
-  token=`cat /tmp/token`
-  curl -s -k --location --request POST ${keycloak_url}'/auth/admin/realms/backstage/groups' \
-	  -H 'Content-Type: application/json' \
-	   -H 'Authorization: Bearer '$token \
-	   --data-raw '{"name": "group'"${0}"'"}'
+function create_group() {
+  token=$(cat $TMP_DIR/token)
+  curl -s -k --location --request POST "$(keycloak_url)/auth/admin/realms/backstage/groups" \
+    -H 'Content-Type: application/json' \
+    -H 'Authorization: Bearer '$token \
+    --data-raw '{"name": "group'"${0}"'"}'
 }
 
-create_groups() {
+function create_groups() {
   echo "Creating Groups in Keycloak"
-  export keycloak_url="https://$(oc get routes keycloak -n ${RHDH_NAMESPACE} -o jsonpath='{.spec.host}')"
-  export keycloak_pass=$(oc -n ${RHDH_NAMESPACE} get secret credential-example-sso -o template --template='{{.data.ADMIN_PASSWORD}}'|base64  -d)
   export -f get_token
-  nohup bash -c  'get_token' &
+  nohup bash -c 'get_token' &
   refresh_pid=$!
   sleep 5
   export -f create_group
-  seq 1 ${GROUP_COUNT}| xargs  -n1 -P10 bash -c 'create_group'
+  seq 1 ${GROUP_COUNT} | xargs -n1 -P10 bash -c 'create_group'
   kill $refresh_pid
 }
 
-create_user() {
-  token=`cat /tmp/token`
-  grp=`echo "${0}%${GROUP_COUNT}"|bc`
-  [[ $grp -eq 0  ]]  && grp=${GROUP_COUNT}
-  curl -s -k --location --request POST ${keycloak_url}'/auth/admin/realms/backstage/users' \
-   -H 'Content-Type: application/json' \
-   -H 'Authorization: Bearer '$token \
-   --data-raw '{"firstName":"test'"${0}"'","lastName":"tester", "email":"test'"${0}"'@test.com", "enabled":"true", "username":"test'"${0}"'","groups":["/group'"${grp}"'"]}'
+function create_user() {
+  token=$(cat $TMP_DIR/token)
+  grp=$(echo "${0}%${GROUP_COUNT}" | bc)
+  [[ $grp -eq 0 ]] && grp=${GROUP_COUNT}
+  curl -s -k --location --request POST "$(keycloak_url)/auth/admin/realms/backstage/users" \
+    -H 'Content-Type: application/json' \
+    -H 'Authorization: Bearer '$token \
+    --data-raw '{"firstName":"test'"${0}"'","lastName":"tester", "email":"test'"${0}"'@test.com", "enabled":"true", "username":"test'"${0}"'","groups":["/group'"${grp}"'"]}'
```

Comment:
Thanks @pmacik  for mixing and sanitizing the code. Will keep these in mind while submitting future patches.